### DISCUSSION TITLE
Monitoring MeshCentral ?

### DISCUSSION THREAD

**Discussion by pierrecorsini** [Ylianst/MeshCentral#5914] [Q&A] [answered]
Hi

I would l like to add to our monitoring dashboards indicators about MeshCentral status and MeshCentral's client status.

For instance: 

- MC uptime.
- Number of client registered.
- Number of client online.
- Number of client offline for more than 1D.

About the availability of the data: everything except the last one seems available (shown on stats panel, or return by console stats commands).

**I do not see auditing API available; is that correct ?**
The only other option I have is to setup scripts on the server that will process the fetching of the data and communication with the monitoring tool. **How would you do this ?** (i.e. bash script to connect with MC database ? dev a direct communication with MC instance ?)

Cya~

**Comment by si458** [comment, marked as answer]
this is correct, the is NO API to get this type of information,
the is already a request open in GitHub issues for 'Number of client offline for more than 1D.' but its a big job to implement!

**Reply by pierrecorsini** [reply]
Thanks for the quick reply.

I'll work on the script based solution.

Any advice on the tech/connector to use to fetch the MC data?

**Reply by si458** [reply]
sadly not I'm still learning all the hidden features that meshcentral hides!
you can run `help` in the server console and get list of commands which you can run and get information back
you can then script these outputs with meshctrl.js

**Comment by simpsonetti** [comment]
hi, don't know if maybe this helps?
https://github.com/amidaware/tacticalrmm

**Reply by pierrecorsini** [reply]
I benchmarked the two and for my usage MeshCentral is perfect. People will argue it's only a remote session tool; but you also have way more management tools (console, terminal, files,...). Just to say I'm fine with MC so far.
